django-backend/fecfiler/scha_transactions/test_models.py

```diff
@@ -49,11 +49,17 @@ def test_generate_uid(self):
         )
         trans = SchATransaction(
             committee_account=committee,
-            report_id=f3x
+            report=f3x
         )
         self.assertFalse(trans.transaction_id)
         trans.generate_unique_transaction_id()
         self.assertTrue(trans.transaction_id)
+        committee.save()
```

Comment:
I think you had the right idea before not testing save() here.  I meant that you should check that the `transaction_id` is assigned on a saved transaction in the `test_save_and_delete` so we can verify `generate_unique_transaction_id` is being utilized correctly in `save()`

Reply:
like at line 72, add `self.assertIsNotNone(transaction_from_db.transaction_id)` this will test that a new transaction will be assigned a transaction_id

Reply:
A check has been added here that expects the saved transaction to have a transaction_id with a length of 20 characters.